extensions/azure-extensions/src/main/java/io/druid/firehose/azure/StaticAzureBlobStoreFirehoseFactory.java

```diff
@@ -0,0 +1,120 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.druid.firehose.azure;
+
+import com.fasterxml.jackson.annotation.JacksonInject;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Charsets;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+import com.metamx.common.CompressionUtils;
+import com.metamx.common.logger.Logger;
+import io.druid.data.input.Firehose;
+import io.druid.data.input.FirehoseFactory;
+import io.druid.data.input.impl.FileIteratingFirehose;
+import io.druid.data.input.impl.StringInputRowParser;
+import io.druid.storage.azure.AzureByteSource;
+import io.druid.storage.azure.AzureStorage;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.io.LineIterator;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+
+public class StaticAzureBlobStoreFirehoseFactory implements FirehoseFactory<StringInputRowParser> {
+  private static final Logger log = new Logger(StaticAzureBlobStoreFirehoseFactory.class);
+
+  private final AzureStorage azureStorage;
+  private final List<AzureBlob> blobs;
+
+  @JsonCreator
+  public StaticAzureBlobStoreFirehoseFactory(
+      @JacksonInject("azureStorage") AzureStorage azureStorage,
+      @JsonProperty("blobs") AzureBlob[] blobs
+  ) {
+    this.azureStorage = azureStorage;
+    this.blobs = ImmutableList.copyOf(blobs);
+  }
+
+  @JsonProperty
+  public List<AzureBlob> getBlobs() {
+    return blobs;
+  }
+
+  @Override
+  public Firehose connect(StringInputRowParser stringInputRowParser) throws IOException {
+    Preconditions.checkNotNull(azureStorage, "null azureStorage");
+
+    final LinkedList<AzureBlob> objectQueue = Lists.newLinkedList(blobs);
+
+    return new FileIteratingFirehose(
+        new Iterator<LineIterator>() {
+          @Override
+          public boolean hasNext() {
+            return !objectQueue.isEmpty();
+          }
+
+          @Override
+          public LineIterator next() {
+            final AzureBlob nextURI = objectQueue.poll();
+
+            final String container = nextURI.getContainer();
+            final String path = nextURI.getPath().startsWith("/")
+                ? nextURI.getPath().substring(1)
+                : nextURI.getPath();
+
+            try {
+              try (final InputStream innerInputStream =
```

Comment:
Can't do this, it will close the inner input stream before the line iterator can go through it.

I hadn't realized this was copied from the S3 static one. I think it will suffice to put it how it was and put a comment at the top of the class that it was inspired by the static s3 firehose.